packages/marshal/src/encodeToSmallcaps.js

```diff
@@ -0,0 +1,461 @@
+   * arrangement, usually by being expressed directly as an object literal.
+   * The exception is copyRecords, whose natural enumeration order
+   * can differ between copyRecords that our distributed object semantics
+   * considers to be equivalent.
+   * Since, for each copyRecord, we only accept string property names,
+   * not symbols, we can canonically sort the names first.
+   * JSON.stringify will then visit these in that sorted order.
+   *
+   * SmallcapsEncoding with a canonical-JSON encoder would also solve
+   * this canonicalness
+   * problem in a more modular and encapsulated manner. Note that the
+   * actual order produced here, though it agrees with canonical-JSON on
+   * copyRecord property ordering, differs from canonical-JSON as a whole
+   * in that the other record properties are visited in the order in which
+   * they are literally written below. TODO perhaps we should indeed switch
+   * to a canonical JSON encoder, and not delicatetly depend on the order
+   * in which these object literals are written.
+   *
+   * Readers must not care about this order anyway. We impose this requirement
+   * mainly to reduce non-determinism exposed outside a vat.
+   *
+   * @param {Passable} passable
+   * @returns {SmallcapsEncoding} except that `encodeToSmallcaps` does not generally
+   * `harden` this result before returning. Rather, `encodeToSmallcaps` is not
+   * directly exposed.
+   * What's exposed instead is a wrapper that freezes the output before
+   * returning. If this turns out to impede static analysis for `harden` safety,
+   * we can always put the (now redundant) hardens back in. They don't hurt.
+   */
+  const encodeToSmallcapsRecur = passable => {
+    // First we handle all primitives. Some can be represented directly as
+    // JSON, and some must be encoded as [SCLASS] composites.
+    const passStyle = passStyleOf(passable);
+    switch (passStyle) {
+      case 'null':
+      case 'boolean': {
+        return passable;
+      }
+      case 'string': {
+        if (passable !== '' && SpecialChars.includes(passable[0])) {
+          return `'${passable}`;
+        }
+        return passable;
+      }
+      case 'symbol': {
+        assertPassableSymbol(passable);
+        const name = /** @type {string} */ (nameForPassableSymbol(passable));
+        return `@${name}`;
+      }
+      case 'undefined': {
+        return '#undefined';
+      }
+      case 'number': {
+        if (Number.isNaN(passable)) {
+          return '#NaN';
+        }
+        if (is(passable, -0)) {
+          return 0;
+        }
+        if (passable === Infinity) {
+          return '#Infinity';
+        }
+        if (passable === -Infinity) {
+          return '#-Infinity';
+        }
+        return passable;
+      }
+      case 'bigint': {
+        const str = String(passable);
+        return passable < 0n ? `${str}n` : `+${str}n`;
+      }
+      case 'copyRecord': {
+        if (hasOwnPropertyOf(passable, SCLASS)) {
```

Comment:
Why not use the existing helper?

```suggestion
        if (hasSClass(passable)) {
```

Reply:
Done